cpp/include/cudf/detail/utilities/cuda.cuh

```diff
@@ -189,6 +189,18 @@ __device__ T single_lane_block_sum_reduce(T lane_value)
   return result;
 }
 
+/**
+ * @brief Get the number of multiprocessors on the device
+ */
+inline cudf::size_type num_multiprocessors()
+{
+  int device = 0;
+  CUDF_CUDA_TRY(cudaGetDevice(&device));
+  int num_sms = 0;
+  CUDF_CUDA_TRY(cudaDeviceGetAttribute(&num_sms, cudaDevAttrMultiProcessorCount, device));
+  return num_sms;
+}
+
```

Comment:
Given that this is a standalone function with no template parameters, could we consider moving the implementation to a .cpp file instead? I think we could create a `src/utilities/cuda.cpp` if there is not a better existing candidate.